### PR TITLE
[ENH] restore accidentally deleted updates to splitter module

### PR DESCRIPTION
**Pull request by fkiraly**
When I merged https://github.com/sktime/sktime/pull/5017, this accidentally reverted the state of the splitters module to about 1 month in the past, as the refactor has not been updated with the updates to the splitter module.

This PR plays back those accidentally deleted updates.

Makes the following additional changes:

* rename of `_config.py` submodule to `_common.py`, as it contains common utility functions - in line with other `_common` modules
* adds author information to each splitter, based on the pre-refactor `git blame` and @fkiraly's memory
* moves tests that are not specific to a splitter but test general base class functionality into a new `test_base` module

### CODE REVIEW

File: sktime/split/base/_base_windowsplitter.py [hunk 1]
```diff
@@ -297,12 +300,38 @@ def get_n_splits(self, y: Optional[ACCEPTED_Y_TYPES] = None) -> int:
...
+
+            y = convert(y, from_type=from_mtype, to_type=to_mtype, as_scitype=scitype)
+
+            index = self._coerce_to_index(y)
+            for _, values in y.groupby(index.droplevel(-1)):
+                # convert to a single ts
+                instance_series = values.reset_index().iloc[:, -2:]
+                instance_series.set_index(instance_series.columns[0], inplace=True)
+                n_splits = len(self.get_cutoffs(instance_series))
+                break
```
Comment (benHeid) [inline sktime/split/base/_base_windowsplitter.py:331]: The `break` statement is reached after the first iteration of the for loop or?

So I suppose the for loop could omitted I think...

**Review by benHeid — CHANGES_REQUESTED** [review]
Thanks, it looks good to me. I have only three questions regarding the code.

File: sktime/split/base/_base_splitter.py [hunk 2]
```diff
@@ -286,7 +339,7 @@ def _check_y(self, y, allow_index=False):
             "pd_multiindex_hier",
         ]
         y_valid, _, y_metadata = check_is_scitype(
-            y, scitype=ALLOWED_SCITYPES, return_metadata=True, var_name="y"
+            y, scitype=ALLOWED_SCITYPES, return_metadata=[], var_name="y"
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply sktime/split/base/_base_splitter.py:342]: if it is an empty list, no metadata are requested, only the check is performed. However, the scitype and mtype is returned in `y_metadata`. Somewhat unclean signature, but this is how it was in the pre-refactor code.

File: sktime/split/base/_base_windowsplitter.py [hunk 3]
```diff
@@ -297,12 +300,38 @@ def get_n_splits(self, y: Optional[ACCEPTED_Y_TYPES] = None) -> int:
...
+
         if y is None:
             raise ValueError(
                 f"{self.__class__.__name__} requires `y` to compute the "
                 f"number of splits."
             )
-        return len(self.get_cutoffs(y))
+
+        multi_scitypes = ["Hierarchical", "Panel"]
+        is_non_single, _, metadata = check_is_scitype(y, multi_scitypes, [])
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply sktime/split/base/_base_windowsplitter.py:312]: see above

File: sktime/split/base/_base_windowsplitter.py (hunk 1 above)
Comment (fkiraly) [reply sktime/split/base/_base_windowsplitter.py:331]: I did not write the code - this is as it was before the refactor. Still, valid comment - should we raise an issue?

Imo, this is just a revert, changes to the code in addition should happen in separate PR.

**Comment by fkiraly** [comment]
> Thanks, it looks good to me. I have only three questions regarding the code.

My reply to these is, could be improved, but the goal of this PR is just to restore deleted code that already got merged.

In that sense, the main criterium for validity should be, "this is exactly as it was before #5017 got merged".

File: sktime/split/temporal_train_test_split.py
Comment (benHeid) [inline sktime/split/temporal_train_test_split.py:1]: I think the examples are missing.

```
  Examples
    --------
    >>> from sktime.forecasting.model_selection import temporal_train_test_split
    >>> from sktime.datasets import load_airline, load_osuleaf
    >>> from sktime.utils._testing.panel import _make_panel
    >>> # univariate time series
    >>> y = load_airline()
    >>> y_train, y_test = temporal_train_test_split(y, test_size=36)
    >>> y_test.shape
    (36,)
    >>> # panel time series
    >>> y = _make_panel(n_instances = 2, n_timepoints = 20)
    >>> y_train, y_test = temporal_train_test_split(y, test_size=5)
    >>> # last 5 timepoints for each instance
    >>> y_test.shape
    (10, 1)
    >>> # time series w/ exogenous variables
    >>> X, y = load_osuleaf(return_X_y=True)
    >>> X_train, X_test, y_train, y_test = temporal_train_test_split(
    ...     X, y, test_size=100)
    >>> X_test.shape, y_test.shape
    ((100, 1), (100,))
```

See: line 1879 and following in sktime/forecasting/model_selection/_split.py before the merge.

File: sktime/split/tests/test_singlewindow.py
Comment (benHeid) [inline sktime/split/tests/test_singlewindow.py:1]: I think 
```python
def test_hierachical_singlewindowsplitter():
    """Test for bugs in SingleWindowSplitter with hierarchical data.
    See #4972
    """
    y = _make_hierarchical(hierarchy_levels=(2, 3), random_state=0)
    splitter = SingleWindowSplitter(fh=[1, 2], window_length=10)
    splits = list(splitter.split(y))
    assert len(splits) == 1, "Should only be one split"


@pytest.mark.parametrize("CV", [SlidingWindowSplitter, ExpandingWindowSplitter])
@pytest.mark.parametrize("fh", [*TEST_FHS, *TEST_FHS_TIMEDELTA])
@pytest.mark.parametrize("window_length", TEST_WINDOW_LENGTHS)
@pytest.mark.parametrize("step_length", TEST_STEP_LENGTHS)
def test_windowbase_splitter_get_n_split_hierarchical(
    CV, fh, window_length, step_length
):
    """Test that WindowBaseSplitter.get_n_splits works for hierarchical data."""
    # see bugs 4971
    y = TEST_Y_PANEL_HIERARCHICAL[0]  # hierachical data
    if _inputs_are_supported([fh, window_length, step_length]):
        cv = CV(fh, window_length, step_length)
        assert cv.get_n_splits(y) == len(
            list(cv.split(y))
        ), "get_n_splits does not equal the number of splits in the output."


@pytest.mark.parametrize("y", TEST_Y_PANEL_HIERARCHICAL)
@pytest.mark.parametrize("CV", [SlidingWindowSplitter, ExpandingWindowSplitter])
def test_windowbase_splitter_get_n_split_unequal_series(y, CV):
    y_unequal = y.copy()  # avoid changing original dataset
    y_unequal.iloc[:3, :] = None  # make the first series shorter than the rest
    y_unequal.dropna(inplace=True)
    cv = CV([1], 24, 1)
    assert cv.get_n_splits(y_unequal) == len(
        list(cv.split(y_unequal))
    ), "get_n_splits does not equal the number of splits in the output."

```

is missing, See line 721 of sktime/forecasting/model_selection/tests/test_split.py before the merge

File: sktime/split/base/_base_windowsplitter.py (hunk 1 above)
Comment (benHeid) [reply sktime/split/base/_base_windowsplitter.py:331]: I aggree

File: sktime/split/temporal_train_test_split.py
Comment (fkiraly) [reply sktime/split/temporal_train_test_split.py:1]: thanks, forgot checking `temporal_train_test_split`, restored

File: sktime/split/tests/test_singlewindow.py
Comment (fkiraly) [reply sktime/split/tests/test_singlewindow.py:1]: thanks, forgot checking the tests! restored

Comment (fkiraly) [reply sktime/split/tests/test_singlewindow.py:1]: (also checked the other tests, could find nothing missing)

**Comment by fkiraly** [comment]
@benHeid, you resolved the two outstanding items but didn't approve, I assume the lack of approval is accidental since you resolved the items.

**Comment by fkiraly** [comment]
(will merge within the next 1-2 hours as @yarnabrina reports some splitter related issues which we should investigate)

**Review by benHeid — APPROVED** [review]

**Pull request merged**